tools/src/main/java/com/hedera/block/tools/commands/BlockInfo.java

```diff
@@ -0,0 +1,327 @@
+                System.out.printf("    Total Items                                = %,d \n", totalItems.get());
+                System.out.printf("    Total Bytes Compressed                     = %,.2f MB\n",
+                        totalBytesCompressed.get() / 1024.0 / 1024.0);
+                System.out.printf("    Total Bytes Uncompressed                   = %,.2f MB\n",
+                        totalBytesUncompressed.get() / 1024.0 / 1024.0);
+                System.out.printf("    Average transactions per block             = %,.2f \n",
+                        totalTransactions.get() / (double) totalBlocks.get());
+                System.out.printf("    Average items per transaction              = %,.2f \n",
+                        totalItems.get() / (double) totalTransactions.get());
+                System.out.printf("    Average uncompressed bytes per transaction = %,d \n",
+                        totalTransactions.get() == 0 ? 0 : (totalBytesUncompressed.get() / totalTransactions.get()));
+                System.out.printf("    Average compressed bytes per transaction   = %,d \n",
+                        totalTransactions.get() == 0 ? 0 : totalBytesCompressed.get() / totalTransactions.get());
+                System.out.printf("    Average uncompressed bytes per item        = %,d \n",
+                        totalItems.get() == 0 ? 0 : totalBytesUncompressed.get() / totalItems.get());
+                System.out.printf("    Average compressed bytes per item          = %,d \n",
+                        totalItems.get() == 0 ? 0 : totalBytesCompressed.get() / totalItems.get());
+                System.out.println("=========================================================");
+            }
+        }
+    }
+
+    /**
+     * Print progress bar to console
+     *
+     * @param progress the progress percentage between 0 and 1
+     */
+    public void printProgress(double progress, int totalBlockFiles, int completedBlockFiles) {
+        final int width = 50;
+        System.out.print("\r[");
+        int i = 0;
+        for (; i <= (int) (progress * width); i++) {
+            System.out.print("=");
+        }
+        for (; i < width; i++) {
+            System.out.print(" ");
+        }
+        System.out.printf("] %.0f%% completed %,d of %,d block files", progress * 100, completedBlockFiles, totalBlockFiles);
+    }
+
+    /**
+     * Collect info for a block file
+     *
+     * @param blockProtoFile the block file to produce info for
+     * @return the info string
+     */
+    public String blockInfo(Path blockProtoFile) {
+        try (InputStream fIn = Files.newInputStream(blockProtoFile)) {
+            byte[] uncompressedData;
+            if (blockProtoFile.getFileName().toString().endsWith(".gz")) {
+                uncompressedData = new GZIPInputStream(fIn).readAllBytes();
+            } else {
+                uncompressedData = fIn.readAllBytes();
+            }
```

Comment:
This branch is currently quite behind, after rebasing you will have access to the `common` module which already has a method that will read file bytes unsafe: `com.hedera.block.common.utils.FileUtilities#readFileBytesUnsafe(Path)` I suggest that you use that.